### PR TITLE
Event handlers need configuration

### PR DESCRIPTION
**Pull request by darrencauthon**
screenshot_123018_103427_am

### CODE REVIEW

File: app/controllers/events_controller.rb [hunk 1]
```diff
@@ -37,4 +37,17 @@ def create_web
     render plain: result.to_json
   end
 
+  def create_with_tree
+
+    originating_data = { url: params[:url] }
+
+    steps = [{ name: 'Apple',  type: 'WebRequest' }]
+
+    workflow = Workflow.build_given_a_hierarchy(first_step: { name: 'Apple', type: 'WebRequest', config: { url: 'http://google.com' }, next_steps: [{ name: 'Orange',  type: 'WebRequest', config: { url: 'http://bing.com' } }, { name: 'Banana',  type: 'WebRequest', config: { url: 'http://yahoo.com' } }] })
```
Comment (calebcauthon) [inline app/controllers/events_controller.rb]: and now we're passing configuration into each event handler.

In this case the configuration is specific to the `type`. the balance of configuration of a type vs a new type is something that will happen based on the types of app we're trying to create. I'm interested to see where types even come from. types made from types sometime, too? and also I'm sure we will string together workflows that can be organized into a single type.

File: app/models/workflow.rb [hunk 2]
```diff
@@ -1,34 +1,74 @@
...
+  def self.set_up_the_method(step, &block)
+    step[:method] = block || lambda { |e| Workflow.build_event_handler_for(step).receive e }
+    step[:config] = {} if step[:config].nil?
+    return if step[:next_steps].nil?
+    step[:next_steps].each { |x| set_up_the_method(x) }
+  end
+
+  def self.build_event_handler_for(step)
+    event_handler = step[:type].constantize.new
+    event_handler.config = step[:config] if event_handler.respond_to?(:config)
```
Comment (calebcauthon) [inline app/models/workflow.rb]: we dont have `def config` anywhere..should this come out til we're using it?

Comment (calebcauthon) [reply app/models/workflow.rb]: oh I see we have it as an accessor NM.

File: app/controllers/events_controller.rb (hunk 1 above)
Comment (darrencauthon) [reply app/controllers/events_controller.rb]: Actually, I'm pretty sure that most of our applications are going to be configuration.  We want the event handlers to be very generic, so generic they are applicable to most scenarios.

**Pull request merged**